Week3/HOMEWORK/js-exercises/exercises-5-totalCost.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+console.log('EXERCISE 5 -  Total cost is ...');
+
+/* EXERCISE-5 =>
+1=> Create a function with one parameter, 
+2=> To add all the numbers in the object, we create a for..in loop,
+3=> Return the total value,
+4=> Create an object with 5 properties,
+5=> Call the function and log it with a string. */
+
+function calculateTotalPrice(shoppingCart) {
+  let total = 0;
+  for (let shoppingItem in shoppingCart) {
+    total += shoppingCart[shoppingItem];
+  }
+  return total;
+}
```

Comment:
This is correct. But I would like to introduce to you two new methods in JavaScript for working with arrays and objects.

**reduce** - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
**Object.keys** - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Using a `for-in` loop is correct but there is a catch to it which you should be aware of. It not only iterates through properties of the current object but also through its parent. This is resolved using `Object.keys`.

For example,

```
const parentParty = {
  cups: 1.5,
  plates: 4,
};

const childParty = Object.create(parent, {
  cola: { value: 3.25, enumerable: true },
  chips: { value: 3.25, enumerable: true },
  gum: { value: 3.25, enumerable: true },
  chocolate: { value: 3.25, enumerable: true },
});

// Ignore the syntax but try using a for-in loop vs Object.keys() now on childParty and see what happens :)
```

Reply:
Thank you for showing other ways. I made this HW in JS1 and we haven't learned reduce yet at that time. We learned this week and I made extra changes with reduce.